src/dotnet/modules/DiffSharpModule/ba.fs

```diff
@@ -0,0 +1,204 @@
+﻿module ba
+
+open System
+open System.Diagnostics
+open System.IO
+
+open DiffSharp.AD.Float64
+open DotnetRunner.Data
+open System.Composition
+open DotnetRunner
+
+let N_CAM_PARAMS = 11
+let ROT_IDX = 0
+let CENTER_IDX = 3
+let FOCAL_IDX = 6
+let X0_IDX = 7
+let RAD_IDX = 9
+
+let dot (a:DV) (b:DV) = a * b
+
+let cross (a:DV) (b:DV) = 
+    toDV [ a.[1] * b.[2] - a.[2] * b.[1]; a.[2] * b.[0] - a.[0] * b.[2]; a.[0] * b.[1] - a.[1] * b.[0] ]
+
+let rodrigues_rotate_point (rot:DV) (x:DV) =
+    let sqtheta = DV.l2normSq rot
+    if sqtheta <> D 0. then
+        let theta = sqrt sqtheta
+        let costheta = cos theta
+        let sintheta = sin theta
+        let theta_inv = D 1. / theta
+
+        let w = rot * theta_inv
```

Comment:
I don't see where theta_inv is used except the line 32. So, maybe it will be more efficient to merge these lines

Reply:
Not sure about efficiency, but I see no reason for `theta_inv`'s existence as well. So merged.